Update/sprints add voyage dates

JoshuaHinman opened the pull request:
# Description

Updates response for route `GET /api/v1/voyages/sprints/teams/{teamId}` - adds startDate and endDate to voyage properties

-also added soloProjectDeadline,  certificateIssueDate,  showcasePublishDate to response body

## Issue link

Fixes task [86b026m0d](https://app.clickup.com/t/86b026m0d)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update


# How Has This Been Tested?
yarn test:e2e sprints

Tested in swagger
![image](https://github.com/chingu-x/chingu-dashboard-be/assets/26824124/b7a28fb9-4e3a-4270-a832-27e64973313c)




# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

cherylli commented on CHANGELOG.md:
```diff
@@ -49,10 +49,12 @@ Another example [here](https://co-pilot.dev/changelog)
 - updated meeting model schema to include optional description field [#135](https://github.com/chingu-x/chingu-dashboard-be/pull/135)
 - Remove teamMeetings from response for getSprintDatesByTeamId [#139](https://github.com/chingu-x/chingu-dashboard-be/pull/139)
 - Updated response for route GET sprints/meetings/{meetingId} to include updatedAt for agendas [#140](https://github.com/chingu-x/chingu-dashboard-be/pull/140)
+- Updated response for route GET /sprints/teams/{teamId}  to include voyage start and end dates [#146](https://github.com/chingu-x/chingu-dashboard-be/pull/146)
```
```suggestion
- Updated response for route GET /sprints/teams/{teamId}  to include voyage start and end dates [#147](https://github.com/chingu-x/chingu-dashboard-be/pull/147)
```

cherylli replied:
not sure how to "suggest change" to move to the bottom

cherylli reviewed (dismissed):
everything works fine but I found out that the endpoint 200 sample response doesn't match the actual response, maybe we can update it in another PR

JoshuaHinman commented:
> everything works fine but I found out that the endpoint 200 sample response doesn't match the actual response, maybe we can update it in another PR

I noticed that too - I could follow up with another revision

cherylli commented:
> > everything works fine but I found out that the endpoint 200 sample response doesn't match the actual response, maybe we can update it in another PR
> 
> I noticed that too - I could follow up with another revision

sure if you don't mind, I've also added another task for that on clickup so if you decide to do it you can claim that and add to this, otherwise happy to just merge this first

Ajen07 reviewed (dismissed):
All changes looks good . It passes all the unit and e2e tests

JoshuaHinman commented:
> > > everything works fine but I found out that the endpoint 200 sample response doesn't match the actual response, maybe we can update it in another PR
> > 
> > 
> > I noticed that too - I could follow up with another revision
> 
> sure if you don't mind, I've also added another task for that on clickup so if you decide to do it you can claim that and add to this, otherwise happy to just merge this first

I've updated the response to match the example, and fixed a test that broke

cherylli reviewed (approved):
thanks. I think you changed the actually response instead of the sample response, but this works too, I think they won't need solo project date but the other 2 dates could be useful in future

JoshuaHinman commented:
> thanks. I think you changed the actually response instead of the sample response, but this works too, I think they won't need solo project date but the other 2 dates could be useful in future

That's funny - yeah I assumed it should go the other way, but whatever you think is fine
